winch/codegen/src/abi/mod.rs

```diff
@@ -60,6 +64,84 @@ pub(super) enum ParamsOrReturns {
     Returns,
 }
 
+/// Macro to get the pinned register holding the [VMContext].
+macro_rules! vmctx {
```

Comment:
This change is fairly large already, so I opted to just introduce this macro and use it in a couple of places that are updated by this change. My plan is to follow up with a general replace after landing this change (but I can do so here too if preferred).